### ISSUE TITLE
cannot pass in spaces with --add-flags with makeBinaryWrapper

### ISSUE THREAD

**Issue by MrQubo** [NixOS/nixpkgs#330471]
### Describe the bug

It seems it's impossible to pass argument that contains space with makeBinaryWrapper. With makeShellWrapper it's possible with `--add-flags '"a b"'`. But makeBinaryWrapper splits it as two arguments `"a` and `b"`.

User `@gerg-l:matrix.org` on matrix proposed to add `--add-flag` to bypass split by space on `--add-flags`.
I'm guessing it might be @Gerg-L?

### Steps To Reproduce

I've made a derivation for testing:

<details>
<summary>default.nix</summary>

```nix
{ stdenv, writeTextFile, makeBinaryWrapper  }:

stdenv.mkDerivation {
  name = "example";

  src = writeTextFile {
    name = "example.c";
    text = ''
      #include <stdio.h>

      int main(int argc, const char const * argv[])
      {
        for (int i = 0; i < argc; i += 1) {
          puts(argv[i]);
        }
      }
    '';
  };
  dontUnpack = true;

  nativeBuildInputs = [ makeBinaryWrapper ];

  buildPhase = ''
    gcc $src -o example
  '';

  installPhase = ''
    install -D example -t $out/bin/
    wrapProgram $out/bin/example --add-flags '"a b"'
  '';

}
```
</details>

<details>
<summary>shell.nix</summary>

```nix
with import <nixpkgs> {};

let
  example = pkgs.callPackage ./default.nix {};

in
pkgs.mkShell {
  packages = [ example ];
}
```
</details>

Run with `nix-shell --run example`.

### Expected behaviour

It should be possible to add/append spaces as arguments to wrapped program.

### Notify maintainers

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Gerg-L** [comment]
Yup it was me

**Comment by eclairevoyant** [comment]
Basically this is a limitation when we don't use `__structuredAttrs`.
See #114044 for a similar issue.

**Comment by MrQubo** [comment]
Hmm, we could add `--add-flag` as a workaround for now, but maybe it's better to wait for https://github.com/NixOS/nixpkgs/pull/318614.

**Comment by eclairevoyant** [comment]
Can you give an example of why this would be useful?

**Comment by MrQubo** [comment]
Whenever you need to overwrite some cli arguments and those arguments contain spaces.
E.g. of arguments that could contain spaces are space-delimited lists or shell commands.

**Comment by Gerg-L** [comment]
> Basically this is a limitation when we don't use `__structuredAttrs`

i don't understand how `__structuredAttrs` would matter at all here
maybe when using `wrapGApps`, but not when using raw `wrapProgram` in the `postInstall`

**Comment by MrQubo** [comment]
You're right, it doesn't matter for the current implementation. Though, with structuredAttrs we could implement makeWrapper differently, using derivation arguments. E.g.

```nix
stdenv.mkDerivation {
...
wrapPrograms = [
  {
    path = "bin/example";
    addFlags = [ "--cmd" "foo bar" ];
  }
];
}
```
